Binary files openssl-orig/Configurations/.unix-Makefile.tmpl.swp and openssl/Configurations/.unix-Makefile.tmpl.swp differ
diff -Nur openssl-orig/Configurations/unix-Makefile.tmpl openssl/Configurations/unix-Makefile.tmpl
--- openssl-orig/Configurations/unix-Makefile.tmpl	2021-08-24 15:38:47.000000000 +0200
+++ openssl/Configurations/unix-Makefile.tmpl	2022-01-12 01:39:07.008659000 +0100
@@ -188,9 +188,14 @@
               $libdir = "lib$multilib";
           }
           file_name_is_absolute($libdir) ? "" : $libdir -}
+LIBDATADIR={- our $libdatadir = $config{libdatadir} || $libdir;
+              file_name_is_absolute($libdatadir) ? "" : $libdatadir -}
 # $(libdir) is chosen to be compatible with the GNU coding standards
 libdir={- file_name_is_absolute($libdir)
           ? $libdir : '$(INSTALLTOP)/$(LIBDIR)' -}
+libdatadir={- file_name_is_absolute($libdatadir)
+              ? $libdatadir : '$(INSTALLTOP)/$(LIBDATADIR)' -}
+pcdir=$(libdatadir)/pkgconfig
 ENGINESDIR=$(libdir)/engines-{- $sover_dirname -}
 
 # Convenience variable for those who want to set the rpath in shared
@@ -654,16 +659,16 @@
 		: {- output_off() if sharedaix(); output_on(); "" -}; \
 	done
 	@ : {- output_on() if $disabled{shared}; "" -}
-	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(libdir)/pkgconfig"
-	@$(ECHO) "install libcrypto.pc -> $(DESTDIR)$(libdir)/pkgconfig/libcrypto.pc"
-	@cp libcrypto.pc "$(DESTDIR)$(libdir)/pkgconfig"
-	@chmod 644 "$(DESTDIR)$(libdir)/pkgconfig/libcrypto.pc"
-	@$(ECHO) "install libssl.pc -> $(DESTDIR)$(libdir)/pkgconfig/libssl.pc"
-	@cp libssl.pc "$(DESTDIR)$(libdir)/pkgconfig"
-	@chmod 644 "$(DESTDIR)$(libdir)/pkgconfig/libssl.pc"
-	@$(ECHO) "install openssl.pc -> $(DESTDIR)$(libdir)/pkgconfig/openssl.pc"
-	@cp openssl.pc "$(DESTDIR)$(libdir)/pkgconfig"
-	@chmod 644 "$(DESTDIR)$(libdir)/pkgconfig/openssl.pc"
+	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(pcdir)"
+	@$(ECHO) "install libcrypto.pc -> $(DESTDIR)$(pcdir)/libcrypto.pc"
+	@cp libcrypto.pc "$(DESTDIR)$(pcdir)"
+	@chmod 644 "$(DESTDIR)$(pcdir)/libcrypto.pc"
+	@$(ECHO) "install libssl.pc -> $(DESTDIR)$(pcdir)/libssl.pc"
+	@cp libssl.pc "$(DESTDIR)$(pcdir)"
+	@chmod 644 "$(DESTDIR)$(pcdir)/libssl.pc"
+	@$(ECHO) "install openssl.pc -> $(DESTDIR)$(pcdir)/openssl.pc"
+	@cp openssl.pc "$(DESTDIR)$(pcdir)"
+	@chmod 644 "$(DESTDIR)$(pcdir)/openssl.pc"
 
 uninstall_dev: uninstall_runtime_libs
 	@$(ECHO) "*** Uninstalling development files"
@@ -703,10 +708,10 @@
 		: {- output_on() unless windowsdll(); "" -}; \
 	done
 	@ : {- output_on() if $disabled{shared}; "" -}
-	$(RM) "$(DESTDIR)$(libdir)/pkgconfig/libcrypto.pc"
-	$(RM) "$(DESTDIR)$(libdir)/pkgconfig/libssl.pc"
-	$(RM) "$(DESTDIR)$(libdir)/pkgconfig/openssl.pc"
-	-$(RMDIR) "$(DESTDIR)$(libdir)/pkgconfig"
+	$(RM) "$(DESTDIR)$(pcdir)/libcrypto.pc"
+	$(RM) "$(DESTDIR)$(pcdir)/libssl.pc"
+	$(RM) "$(DESTDIR)$(pcdir)/openssl.pc"
+	-$(RMDIR) "$(DESTDIR)$(pcdir)"
 	-$(RMDIR) "$(DESTDIR)$(libdir)"
 
 install_engines: install_runtime_libs build_engines
diff -Nur openssl-orig/Configure openssl/Configure
--- openssl-orig/Configure	2021-08-24 15:38:47.000000000 +0200
+++ openssl/Configure	2022-01-12 01:25:09.839253000 +0100
@@ -314,6 +314,7 @@
 $config{openssldir}="";
 $config{processor}="";
 $config{libdir}="";
+$config{libdatadir}="";
 my $auto_threads=1;    # enable threads automatically? true by default
 my $default_ranlib;
 
@@ -814,6 +815,10 @@
                 elsif (/^--libdir=(.*)$/)
                         {
                         $config{libdir}=$1;
+                        }
+                elsif (/^--libdatadir=(.*)$/)
+                        {
+                        $config{libdatadir}=$1;
                         }
                 elsif (/^--openssldir=(.*)$/)
                         {
